src/input/mod.rs

```diff
@@ -281,7 +284,12 @@ impl State {
                     let serial = SERIAL_COUNTER.next_serial();
                     let time = Event::time_msec(&event);
                     let keyboard = seat.get_keyboard().unwrap();
-                    let current_focus = keyboard.current_focus();
+                    let mut current_focus = keyboard.current_focus();
+                    if self.common.session_lock.is_some() {
+                        if !matches!(current_focus, Some(KeyboardFocusTarget::LockSurface(_))) {
+                            current_focus = None;
+                        }
+                    }
```

Comment:
Could we rather fix this in `Shell::refresh_focus` and force the focus to be on the lock surface?

Reply:
I'm not entirely sure about the best way to handle keyboard focus, but there are a few things to consider:
* If the session is locked but there is are no lock surfaces (the session lock client has not created them yet, or has crashed), no surface should receive keyboard events.
* If there are multiple outputs, each of them have lock surfaces that can receive keyboard input, so it makes sense to handle changes to keyboard focus normally, but only allow lock surfaces to have focus.
* The session is locked and lock surfaces are created, the lock surface on the current output should gain initial focus.

Looking at `refresh_focus`, that could be a good place to handle this. I think the first and last of these things could be handled there, while it could avoid messing with the surface by no changing focus when it's already an alive lock surface.